pkg/client/metrics/metrics.go

```diff
@@ -18,30 +18,45 @@ package metrics
 
 import (
 	"context"
+	"net/http"
 	"net/url"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	"k8s.io/component-base/metrics/legacyregistry"
+	apimetrics "k8s.io/apiserver/pkg/endpoints/metrics"
+	"k8s.io/component-base/metrics"
 	"sigs.k8s.io/prometheus-adapter/pkg/client"
 )
 
 var (
 	// queryLatency is the total latency of any query going through the
 	// various endpoints (query, range-query, series).  It includes some deserialization
 	// overhead and HTTP overhead.
-	queryLatency = prometheus.NewHistogramVec(
-		prometheus.HistogramOpts{
-			Name:    "cmgateway_prometheus_query_latency_seconds",
-			Help:    "Prometheus client query latency in seconds.  Broken down by target prometheus endpoint and target server",
-			Buckets: prometheus.ExponentialBuckets(0.0001, 2, 10),
+	queryLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Namespace: "prometheus_adapter",
+			Subsystem: "prometheus_client",
+			Name:      "request_duration_seconds",
+			Help:      "Prometheus client query latency in seconds.  Broken down by target prometheus endpoint and target server",
+			Buckets:   prometheus.DefBuckets,
 		},
 		[]string{"endpoint", "server"},
```

Comment:
`endpoint` is already a target label. If `honor_labels` is false (which is the default), the metric label will be renamed to `exported_endpoint` to preserve the target label.
Since the metric was never exposed before, I'd suggest using `path` instead.
```suggestion
		[]string{"path", "server"},
```

Reply:
yeah path would be a good fit since the values can be one of these: https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/pkg/client/api.go#L139-L141